packages/daemon/src/runner/helper.ts

```diff
@@ -143,3 +157,60 @@ export function markFailures(components: PipcookComponentResult[]): PipcookCompo
     return nextComponent;
   });
 }
+
+export async function parseConfig(configPath: string, generateId = true) {
+  const configJson: RunConfigI = await fs.readJson(configPath);
+  const result: PipelineDB = {};
+  if (generateId) {
+    result.id = uuidv1();
+  }
+
+  if (configJson.name) {
+    result.name = configJson.name;
+  }
+
+  PLUGINS.forEach((pluginType) => {
+    if (configJson.plugins[pluginType] &&
+      configJson.plugins[pluginType].package &&
+        LifeCycleTypes[pluginType]) {
+      const pluginName = configJson.plugins[pluginType].package;
+      const params = configJson.plugins[pluginType].params || {};
+
+      result[pluginType] = pluginName;
+      const paramsAttribute: PipelineDBParams = (pluginType + 'Params') as PipelineDBParams;
+      result[paramsAttribute] = JSON.stringify(params);
+    }
+  });
+
+  return result;
+}
+
+export async function createRun(pipelineId: string): Promise<RunDB> {
+  const packageJson = await fs.readJSON(path.join(__dirname, '..', '..', 'package.json'));
+  return {
+    id: uuidv1(),
+    pipelineId,
+    coreVersion: packageJson.version,
+    status: PipelineStatus.INIT,
+    currentIndex: -1
+  };
+}
+
+export async function writeOutput(jobId: string, content: string, stderr = false) {
+  const fileName = stderr ? 'stderr' : 'stdout';
+  const filePath = path.join(PIPCOOK_LOGS, jobId, fileName);
+  await new Promise((resolve, reject) => {
```

Comment:
does fs.appendFile return Promise?

Reply:
it seems not. Or use util.promisify

Reply:
It does return `Promise` without callback as a parameter :P

`fx-extra` wrapper the API from `fs` via `universalify` to support `Promise`
https://github.com/jprichardson/node-fs-extra/blob/master/lib/fs/index.js#L58-L61

Reply:
Oh, thanks :). Forgot that we are using fs-extra